src/components/Molecules/Membership/Form/Form.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import Text from '../../../Atoms/Text/Text';
+import MoneyBox from '../MoneyBox/MoneyBox';
+import {
+  handleDonateSubmission,
+  onKeyPress,
+  isAmountValid,
+  isInputMatchBoxValue
+} from '../../../../utils/Membership';
+import {
+  Button,
+  Copy,
+  Error,
+  FormFieldset,
+  FormWrapper,
+  Label,
+  Form,
+  MoneyBuys,
+  AmountField
+} from '../Membership.style';
+
+const Signup = ({
+  data: { regularGiving },
+  otherDescription,
+  clientID,
+  cartID,
+  mbshipID,
+  ...rest
+}) => {
+  const [moneyBoxes, setMoneyBoxes] = useState({
+    box1: '',
+    box2: '',
+    box3: ''
+  });
+
+  const [isSelected, setIsSelected] = useState(true); // Highlight one the money buy box when the page load
+  const [errorMsg, setErrorMsg] = useState(false);
+  const [amountDonate, setAmountDonate] = useState('');
+  const [userInput, setUserInput] = useState('');
+  const [boxBorderColor, setBoxBorderColor] = useState('');
+  const [inputBorderColor, setInputBorderColor] = useState(false);
+  const [moneyBuyCopy, setMoneyBuyCopy] = useState(true);
+
+  // This function is triggered when one of the money buy box is selected
+  const selectMoneyBuy = (copy, value) => {
+    if (isSelected) setIsSelected(false);
+    if (errorMsg) setErrorMsg(false);
+    // Check if input is highlighted and his value matches one of the money buy box
+    const isUserInputMatch = userInput * 1 === value; // convert string to number string * 1
+    if (inputBorderColor && !isUserInputMatch) {
+      setInputBorderColor(false);
+      setUserInput('');
+    }
+    setBoxBorderColor(value);
+    setMoneyBuyCopy(copy);
+    setAmountDonate(value);
+  };
+
+  // Handle user other amount input
+  const handleChange = (input, description) => {
+    if (!isAmountValid(input)) {
+      if (moneyBuyCopy) setMoneyBuyCopy(false);
+      if (boxBorderColor) setBoxBorderColor(false);
+      if (!errorMsg) setErrorMsg(true);
+    } else {
+      setBoxBorderColor(false);
+      if (errorMsg) setErrorMsg(false);
+      setMoneyBuyCopy(description);
+      setAmountDonate(input);
+    }
+    // Highlight both input and box
+    isInputMatchBoxValue(moneyBoxes, selectMoneyBuy, setAmountDonate, input);
+    setInputBorderColor(true);
+    setUserInput(input);
+  };
+
+  const hightlightInput = (value, description) => {
+    if (isSelected) setIsSelected(false);
+    if (errorMsg) {
+      setMoneyBuyCopy(false);
+    } else if (!value) {
+      setAmountDonate(0);
+      setBoxBorderColor(false);
+      setMoneyBuyCopy(description);
+    }
+    setInputBorderColor(true);
+  };
+
+  useEffect(() => {
+    regularGiving.moneybuys.map((moneyBuy, index) => {
+      const box = `box${index + 1}`;
+      // eslint-disable-next-line no-shadow
+      setMoneyBoxes(moneyBoxes => ({ ...moneyBoxes, [box]: moneyBuy }));
+      return (
+        isSelected &&
+        index === 1 &&
```

Comment:
I believe `index === 1` that what makes the second box, selected by default, i am not sure if there's no flexibility required here, for instance, if we wanted more than three boxes, or default one to be something other than the second one, is that an actual concern ?

Reply:
Yep, it selects the second box, yeah maybe later if we plan to add more boxes we could set field (bool) within contentful to control that... or something better